src/loader/bgw_launcher.c

```diff
@@ -610,6 +613,27 @@ ts_bgw_cluster_launcher_main(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+/*
+ * Check whether a database is a template database and raise an error if so as we don't want to run in template dbs.
+ * Check in the syscache rather than searching through the entire database catalog again.
```

Comment:
1) Can you move this comment to above the call to `database_is_template_check` inside `ts_bgw_db_scheduler_entrypoint` below?
2) Can you update this comment to say something about how we need to check inside the entrypoint function for the case when `restart_background_workers` is called, in which case the DB will try to restart a scheduler, even for a DB that didn't have a working scheduler to begin with? (ie the case where you call `create extension` within a template). Otherwise, as is, it's unclear why you need this if you are pruning away templates in `populate_database_htab`.

Reply:
Good point, def need to mention that. 
I think I'd prefer the other way round (ie this comment stays where it is, describing what the function does and comment about why we have to call at this point in time goes at the point of call) as that makes more sense to me, but if you feel strongly I can change where each live. lmk

Reply:
Done.